Quarkus NativeImageBuildStep fails with perm denied with podman 2.x

**Describe the bug**
This bug originated here:
https://bugzilla.redhat.com/show_bug.cgi?id=1854738
and here:
https://groups.google.com/g/quarkus-dev/c/8LVoqk4G5uc/m/M5MAVopLAwAJ

Basically with podman 1.x --userns=keep-id contained a bug which got fixed with 2.x which results in quarkus build failures with options `-Pnative -Dquarkus.native.container-build=true -Dquarkus.native.container-runtime=podman` in rootless mode.

**Expected behavior**
Native image build passes. No "Permission Denied" error.

**Actual behavior**
Native image build in the container with mounted volume fails due to "Permission Denied".

**To Reproduce**
Steps to reproduce the behavior:
1. Install podman 2.0 (for example in F32)
2. `mvn package -Pnative -Dquarkus.native.container-build=true -Dquarkus.native.container-runtime=podman`

**Additional context**
Looking at this podman bug:
https://github.com/containers/podman/issues/3990

... it seems podman won't help with getting permissions correctly set up for volume mounts.

My experiments showed that doing some `podman unshare` tricks prior the native image build might work. Something like this:

```
$ CONTAINER_UID=$(podman run --rm --entrypoint /bin/bash -ti quay.io/quarkus/ubi-quarkus-native-image:19.3.1-java11 -c 'id -u' | sed 's/\r//g')
$ podman unshare chown -R $CONTAINER_UID target/getting-started-1.0-SNAPSHOT-native-image-source-jar
$ podman run -v getting-started/target/getting-started-1.0-SNAPSHOT-native-image-source-jar:/project:z --env LANG=C --rm quay.io/quarkus/ubi-quarkus-native-image:19.3.1-java11 -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -J-Duser.language=en -J-Dfile.encoding=UTF-8 --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy\$BySpaceAndTime -H:+JNI -jar getting-started-1.0-SNAPSHOT-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:-IncludeAllTimeZones -H:EnableURLProtocols=http --no-server -H:-UseServiceLoaderFeature -H:+StackTrace getting-started-1.0-SNAPSHOT-runner
```
Getting the UID the container runs as that way is important as it might change.

FWIW, I get the same error with docker too:

```
 mvn clean verify -Pnative -Dquarkus.native.container-build=true
```
fails during the *IT test runs when it tries to launch the native application for testing:

```
[INFO] Running org.acme.NativeExampleResourceIT
Executing [code-with-quarkus/target/code-with-quarkus-1.0.0-SNAPSHOT-runner, -Dquarkus.http.port=8081, -Dquarkus.http.ssl-port=8444, -Dtest.url=http://localhost:8081, -Dquarkus.log.file.path=target/target/quarkus.log]
code-with-quarkus/target/code-with-quarkus-1.0.0-SNAPSHOT-runner: code-with-quarkus/target/code-with-quarkus-1.0.0-SNAPSHOT-runner: cannot execute binary file

...
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.251 s <<< FAILURE! - in org.acme.NativeExampleResourceIT
[ERROR] org.acme.NativeExampleResourceIT  Time elapsed: 0.25 s  <<< ERROR!
org.junit.platform.commons.JUnitException: Quarkus native image start failed, original cause: java.lang.RuntimeException: Failed to start native image, process has exited
Caused by: java.lang.RuntimeException: Failed to start native image, process has exited
```
So this doesn't look specific to podman and impacts docker too. Here's the output of `docker version` command:

```
Client: Docker Engine - Community
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        afacb8b
 Built:             Wed Mar 11 01:21:11 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.17
  Git commit:       afacb8b
  Built:            Wed Mar 11 01:29:16 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java
@@ -101,15 +101,14 @@ public class NativeImageBuildStep {
                     outputPath + ":" + CONTAINER_BUILD_VOLUME_PATH + ":z", "--env", "LANG=C");
 
             if (SystemUtils.IS_OS_LINUX) {
-                if ("docker".equals(containerRuntime)) {
-                    String uid = getLinuxID("-ur");
-                    String gid = getLinuxID("-gr");
-                    if (uid != null && gid != null && !"".equals(uid) && !"".equals(gid)) {
-                        Collections.addAll(nativeImage, "--user", uid + ":" + gid);
+                String uid = getLinuxID("-ur");
+                String gid = getLinuxID("-gr");
+                if (uid != null && gid != null && !"".equals(uid) && !"".equals(gid)) {
+                    Collections.addAll(nativeImage, "--user", uid + ":" + gid);
+                    if ("podman".equals(containerRuntime)) {
+                        // Needed to avoid AccessDeniedExceptions
+                        nativeImage.add("--userns=keep-id");
                     }
-                } else if ("podman".equals(containerRuntime)) {
-                    // Needed to avoid AccessDeniedExceptions
-                    nativeImage.add("--userns=keep-id");
                 }
             }
             nativeConfig.containerRuntimeOptions.ifPresent(nativeImage::addAll);

